lib/rules/jsx-max-depth.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @fileoverview Validate JSX maximum depth
+ * @author Chris<wfsr@foxmail.com>
+ */
+'use strict';
+
+let variableUtil = require('../util/variable');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Validate JSX maximum depth',
+      category: 'Stylistic Issues',
+      recommended: false
+    },
+    schema: [{
+      oneOf: [
+        {
+          type: 'integer',
+          minimum: 0
+        },
+        {
+          type: 'object',
+          properties: {
+            max: {
+              type: 'integer',
+              minimum: 0
+            }
+          }
+        }
+
+      ]
+    }]
+  },
+  create: function(context) {
+    const MESSAGE = 'Expected the depth of JSX Elements nested should be {{needed}} but found {{gotten}}.';
+    const DEFAULT_DEPTH = 3;
+
+    let option = context.options[0] || DEFAULT_DEPTH;
+    let maxDepth = (typeof option === 'number' ? option : option.max) || DEFAULT_DEPTH;
```

Comment:
`const`

Also, what happens if I pass a value of `0`? This looks like it will always set 0 to 3. Please add some tests for values of `0` - ie, no nesting allowed.